thorclient/bind/contract.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2025 The VeChainThor developers
+//
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package bind
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/vechain/thor/v2/thor"
+	"github.com/vechain/thor/v2/thorclient"
+)
+
+// Contract is the main interface for contract interactions.
+// It provides a unified entry point for all contract operations.
+type Contract interface {
+	// Method creates a new method builder for the specified method and arguments.
+	Method(method string, args ...any) MethodBuilder
```

Comment:
```suggestion
	Method(method string) MethodBuilder
```
Arguments can be passed for building the call/send builder
```
method := contract.MethodBuilder("transfer", to, amount)

method.Call(to, amount).Execute()
method.Send(to, amount).Send()
```